### PR TITLE
Making VisibilityTimeout configurable (#844)

### PR DESCRIPTION
**Pull request by mathewc**
Addresses https://github.com/Azure/azure-webjobs-sdk/issues/844

I'll be exposing this in Functions as well.

### CODE REVIEW

File: sample/SampleHost/Program.cs [hunk 1]
```diff
@@ -10,6 +11,8 @@ class Program
         static void Main(string[] args)
         {
             var config = new JobHostConfiguration();
+            config.Queues.VisibilityTimeout = TimeSpan.FromSeconds(15);
```
Comment (mathewc) [inline sample/SampleHost/Program.cs]: For my local testing. With these changes, when the sample function throws an exception, there is a 15 second delay before the next retry now.

File: src/Microsoft.Azure.WebJobs.Host/Queues/Listeners/QueueListener.cs [hunk 2]
```diff
@@ -73,6 +74,10 @@ internal sealed class QueueListener : IListener, ITaskSeriesCommand, INotificati
             _trace = trace;
             _queueConfiguration = queueConfiguration;
 
+            // if the function runs longer than this, the invisibility will be updated
+            // on a timer periodically for the duration of the function execution
+            _visibilityTimeout = TimeSpan.FromMinutes(10);
```
Comment (mathewc) [inline src/Microsoft.Azure.WebJobs.Host/Queues/Listeners/QueueListener.cs]: Note that this existing internal value is still not configurable, and I don't think it needs to be.

**Comment by mathewc** [comment]
I'll add a test

**Comment by markanewman** [comment]
I see the sample for WebJobs above, but I don't see one for Functions. Can we get a link to that PR here too?

**Pull request closed without merging**